protocol/storage/miner.go

```diff
@@ -601,19 +613,46 @@ func (sm *Miner) OnCommitmentAddedToChain(sector *sectorbuilder.SealedSectorMeta
 }
 
 func (sm *Miner) onCommitSuccess(dealCid cid.Cid, sector *sectorbuilder.SealedSectorMetadata) {
-	err := sm.updateDealResponse(dealCid, func(resp *storagedeal.Response) {
+	pieceInfo, err := sm.findPieceInfo(dealCid, sector)
+	if err != nil {
+		// log error, but continue to update deal with the information we have
+		log.Errorf("commit succeeded, but could not find piece info %s", err)
+	}
+
+	// failure to locate commitmentMessage should not block update
+	commitmentMessage := sm.dealsAwaitingSeal.CommitmentMessages[sector.SectorID]
+
+	err = sm.updateDealResponse(dealCid, func(resp *storagedeal.Response) {
+
 		resp.State = storagedeal.Posted
 		resp.ProofInfo = &storagedeal.ProofInfo{
-			SectorID: sector.SectorID,
-			CommR:    sector.CommR[:],
-			CommD:    sector.CommD[:],
+			SectorID:          sector.SectorID,
+			CommitmentMessage: &commitmentMessage,
+		}
+		if pieceInfo != nil {
+			resp.ProofInfo.PieceInclusionProof = pieceInfo.InclusionProof
 		}
 	})
 	if err != nil {
 		log.Errorf("commit succeeded but could not update to deal 'Posted' state: %s", err)
 	}
 }
 
+// search the sector's piece info to find the one for the given deal's piece
+func (sm *Miner) findPieceInfo(dealCid cid.Cid, sector *sectorbuilder.SealedSectorMetadata) (*sectorbuilder.PieceInfo, error) {
+	deal := sm.porcelainAPI.DealGet(dealCid)
+	if deal == nil || deal.Response.State == storagedeal.Unknown {
+		return nil, errors.Errorf("Could not find deal with deal cid %s", dealCid)
+	}
+
+	for _, info := range sector.Pieces {
+		if info.Ref == deal.Proposal.PieceRef {
```

Comment:
BLOCKING: You should use `info.Ref.Equals(deal.Proposal.PieceRef)` instead of `==` (in spite of the fact that `Equals` delegates to `==`). Maybe I'm being overly paranoid, but it is possible that the `go-cid` maintainers change the implementation out from underneath you and `==` no longer does what you'd expect.